dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/navigation/NavToolTest.java

```diff
@@ -409,4 +404,41 @@ private int findShowOnMenuUnderFolder(Folder folder, User user)
         return showOnMenu;
     }
 
+    @Test
+    public void testNavTool_rootLevel_returnItemsOnlyForSite() throws Exception
+    {
+        //Get SystemFolder
+        final Folder systemFolder = APILocator.getFolderAPI().findSystemFolder();
+
+        //Create new Host
+        final Host newHost1 = new SiteDataGen().nextPersisted();
+
+        //Create contentlets on one host
+        final File file = File.createTempFile("fileTestEngTrue", ".txt");
+        FileUtil.write(file, "helloworld");
+        final Contentlet fileAssetOneHost = new FileAssetDataGen(systemFolder, file)
+                .host(newHost1).setProperty(FileAssetAPI.SHOW_ON_MENU, "true").nextPersisted();
+        final Template template = new TemplateDataGen().nextPersisted();
+        final Contentlet pageAssetOneHost = new HTMLPageDataGen(systemFolder, template)
+                .showOnMenu(true).host(newHost1).nextPersisted();
+        ContentletDataGen.publish(pageAssetOneHost);
+        ContentletDataGen.publish(fileAssetOneHost);
+
+        //Create new Host
+        final Host newHost2 = new SiteDataGen().nextPersisted();
+
+        //Create contentlets on default host
+        final Contentlet fileAssetNewHost = new FileAssetDataGen(systemFolder, file)
+                .host(newHost2).setProperty(FileAssetAPI.SHOW_ON_MENU, "true").nextPersisted();
+        final Contentlet pageAssetNewHost = new HTMLPageDataGen(systemFolder, template)
+                .showOnMenu(true).host(newHost2).nextPersisted();
+        ContentletDataGen.publish(pageAssetNewHost);
+        ContentletDataGen.publish(fileAssetNewHost);
+
+        NavResult navResult = new NavTool().getNav(newHost2, systemFolder.getPath());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'navResult' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4828419)